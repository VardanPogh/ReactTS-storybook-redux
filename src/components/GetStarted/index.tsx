import React from 'react'
import { useHistory } from 'react-router-dom'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import clsx from 'clsx'
import LinkMaterialUI from '@material-ui/core/Link'
import Button from '../Button'
import InputField from '../InputField'

const useStyles = makeStyles(() =>
  createStyles({
    section: {
      marginBottom: 20,
    },
    sectionWithBackground: {
      backgroundImage: 'url(img/bg.png)',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      height: 1200,
      width: '100%',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      paddingBottom: 30,
      '@media (max-width: 1024px)': {
        height: 900,
      },
      '@media (max-width: 768px)': {
        height: 'auto',
        background: 'none',
        backgroundColor: '#BED9F7',
      },
    },
    getStarted: {
      textAlign: 'center',
      '& h2': {
        color: '#fff',
      },
    },
    link: {
      cursor: 'pointer',
    },
  })
)

const GetStarted: React.FC = () => {
  const classes = useStyles()
  const history = useHistory()
  return (
    <section
      className={clsx(
        classes.section,
        classes.sectionWithBackground,
        classes.getStarted
      )}
    >
      <Container maxWidth="lg">
        <h2>Get Started</h2>
        <p>
          No obligation. Preview before purchase. Money back guarantee. Just 5
          min.
        </p>
        <InputField type="text" title="Enter email address" />
        <Button
          text="Get Started"
          variant="contained"
          onClick={(): void => history.push('/battery_use')}
        />
        <div style={{ margin: '20px 0' }}>
          <LinkMaterialUI
            className={classes.link}
            onClick={(): void => history.push('/battery_use')}
          >
            Or Get Started without an email address here.
          </LinkMaterialUI>
        </div>
      </Container>
    </section>
  )
}

export default GetStarted
