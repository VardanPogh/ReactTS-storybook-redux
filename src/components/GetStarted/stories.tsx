import React from 'react'
import GetStarted from '.'

export default {
  title: 'GetStarted',
  component: GetStarted,
}

export const Default: React.FC = () => <GetStarted />
