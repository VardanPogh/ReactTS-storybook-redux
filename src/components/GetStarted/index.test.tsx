import React from 'react'
import { render } from '@testing-library/react'
import GetStarted from '.'

describe('GetStarted Test', () => {
  test('Render GetStarted', () => {
    const { container } = render(<GetStarted />)
    expect(container).toMatchSnapshot()
  })
})
