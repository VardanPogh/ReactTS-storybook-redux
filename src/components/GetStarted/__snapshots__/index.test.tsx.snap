// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted Test Render GetStarted 1`] = `
<div>
  <section
    class="makeStyles-section-1 makeStyles-sectionWithBackground-2 makeStyles-getStarted-3"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <h2>
        Get Started
      </h2>
      <p>
        No obligation. Preview before purchase. Money back guarantee. Just 5 min.
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Enter email address
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-8"
            >
              <span>
                Enter email address
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained-10"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Get Started
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        style="margin: 20px 0px;"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-4 MuiTypography-colorPrimary"
        >
          Or Get Started without an email address here.
        </a>
      </div>
    </div>
  </section>
</div>
`;
